### PR TITLE
tests: use single ServiceEntry with multiple addresses in ambient tests

### PR DESCRIPTION
**Pull request by jewertow**
**Please provide a description of this PR:**

It's a follow-up to https://github.com/istio/istio/pull/51939 and addresses this comment: https://github.com/istio/istio/pull/51939#discussion_r1668976103.

### CODE REVIEW

File: tests/integration/ambient/baseline_test.go [hunk 1]
```diff
@@ -1852,38 +1852,16 @@ spec:
         host: "{{.Destination}}"
 `).ApplyOrFail(t)
 
-			// TODO(https://github.com/istio/istio/issues/51747) use a single SE instead of one for v4 and one for v6
 			cfg := config.YAML(`
-{{ $to := .To }}
 apiVersion: networking.istio.io/v1beta1
 kind: ServiceEntry
 metadata:
   name: test-se-v4
```
Comment (howardjohn) [inline tests/integration/ambient/baseline_test.go:1859]: nit

```suggestion
  name: test-se
```

File: tests/integration/ambient/baseline_test.go [hunk 2]
```diff
@@ -1921,22 +1899,16 @@ spec:
...
 								// TODO validate L7 processing/some headers indicating we reach the svc we wanted
-								if v4 {
-									from.CallOrFail(t, echo.CallOptions{
-										Address: "240.240.240.255",
-										Port:    to.PortForName("http"),
-										// If request is sent before service is processed it will hit 10s timeout, so fail faster
-										Timeout: time.Millisecond * 500,
-									})
-								}
+								address := "240.240.240.255"
```
Comment (howardjohn) [inline tests/integration/ambient/baseline_test.go:1902]: I think we still want to do both? Right now we only validate one or the other works

**Review by howardjohn — COMMENTED** [review]
Thanks!

File: tests/integration/ambient/baseline_test.go (hunk 2 above)
Comment (jewertow) [reply tests/integration/ambient/baseline_test.go:1902]: Yes, I remember that you suggested that and replied in the original [comment](https://github.com/istio/istio/pull/51939#discussion_r1684902590). It does not work for me, so I am not sure if I correctly understand your suggestion. How we can use both addresses here?

Comment (howardjohn) [reply tests/integration/ambient/baseline_test.go:1902]: The exact code as before.

```
if v6 { call v6 }
if v4 { call v4}
```

vs what we have now
```
if v6 { call v6 }
else { call v4}
```

File: tests/integration/ambient/baseline_test.go (hunk 1 above)
Comment (jewertow) [reply tests/integration/ambient/baseline_test.go]: Thanks, done.

File: tests/integration/ambient/baseline_test.go (hunk 2 above)
Comment (jewertow) [reply tests/integration/ambient/baseline_test.go]: Ah, right!

File: tests/integration/ambient/waypoint_test.go [hunk 3]
```diff
@@ -381,21 +381,33 @@ func TestWaypointDNS(t *testing.T) {
 				if src.Config().HasSidecar() {
 					t.Skip("TODO: sidecars don't properly handle use-waypoint")
 				}
-				address := "240.240.240.239"
-				if _, v6 := getSupportedIPFamilies(t); v6 {
-					address = "2001:2::f0f0:239"
+				v4, v6 := getSupportedIPFamilies(t)
+				if v4 {
+					src.CallOrFail(t, echo.CallOptions{
```
Comment (howardjohn) [inline tests/integration/ambient/waypoint_test.go:386]: Long term, what we may want to do is add an option into echo.CallOptions which is either (or both):

* `Addresses []string`: call each one
* DualStack=true: call the address, but do a dns lookup only for V4 then only for V6.

With the auto allocation v2 work @ilrudie is doing/just merged, we can actually call the hostname here and do the DNS lookup.

Then its simple to call both

Comment (howardjohn) [reply tests/integration/ambient/waypoint_test.go:386]: Not for this PR, just long term -- ideally we get to the point we are doing dual stack in every test, so making it simple is probably increasingly the likelihood it actually gets tested

**Review by howardjohn — APPROVED** [review]
:+1: :+1:

**Comment by jewertow** [comment]
It fails when request is sent from sidecar:
```
=== RUN   TestServiceEntryInlinedWorkloadEntry/MESH_INTERNAL_STATIC_0/sidecar
2024-07-19T20:00:04.502045Z	info	tf	=== BEGIN: Test: 'ambient[TestServiceEntryInlinedWorkloadEntry/MESH_INTERNAL_STATIC_0/sidecar]' ===
=== RUN   TestServiceEntryInlinedWorkloadEntry/MESH_INTERNAL_STATIC_0/sidecar/to_captured
2024-07-19T20:00:04.505984Z	info	tf	=== BEGIN: Test: 'ambient[TestServiceEntryInlinedWorkloadEntry/MESH_INTERNAL_STATIC_0/sidecar/to_captured]' ===
    baseline_test.go:1911: 2 errors occurred:
        	* failed calling sidecar (cluster=cluster-0)->'http://[2001:2::f0f0:255]:80': call failed from sidecar (cluster=cluster-0) to http://[2001:2::f0f0:255]:80 (using http): expected no error, but encountered rpc error: code = Unknown desc = request set timed out after 500ms and only 0/1 requests completed (0s avg)
        	* failed calling sidecar (cluster=cluster-0)->'http://[2001:2::f0f0:255]:80': call failed from sidecar (cluster=cluster-0) to http://[2001:2::f0f0:255]:80 (using http): expected no error, but encountered rpc error: code = Unknown desc = request set timed out after 500ms and only 0/1 requests completed (0s avg)
```
I have to quickly add support for matching multiple VIPs in HTTP routes. So I am reverting this change from baseline_test and will add it again in the PR with HTTP routes.

**Pull request merged**